permissions/resource_permissions.go

```diff
@@ -108,315 +28,137 @@ type PermissionsAPI struct {
 	context context.Context
 }
 
-func isDbsqlPermissionsWorkaroundNecessary(objectID string) bool {
-	return strings.HasPrefix(objectID, "/sql/") && !strings.HasPrefix(objectID, "/sql/warehouses")
-}
-
-func urlPathForObjectID(objectID string) string {
-	if isDbsqlPermissionsWorkaroundNecessary(objectID) {
-		// Permissions for SQLA entities are routed differently from the others.
-		return "/preview/sql/permissions" + objectID[4:]
-	}
-	return "/permissions" + objectID
-}
-
-// As described in https://github.com/databricks/terraform-provider-databricks/issues/1504,
-// certain object types require that we explicitly grant the calling user CAN_MANAGE
-// permissions when POSTing permissions changes through the REST API, to avoid accidentally
-// revoking the calling user's ability to manage the current object.
-func (a PermissionsAPI) shouldExplicitlyGrantCallingUserManagePermissions(objectID string) bool {
-	for _, prefix := range [...]string{"/registered-models/", "/clusters/", "/instance-pools/", "/serving-endpoints/", "/queries/", "/sql/warehouses"} {
-		if strings.HasPrefix(objectID, prefix) {
-			return true
-		}
-	}
-	return isDbsqlPermissionsWorkaroundNecessary(objectID)
-}
-
-func isOwnershipWorkaroundNecessary(objectID string) bool {
-	return strings.HasPrefix(objectID, "/jobs") || strings.HasPrefix(objectID, "/pipelines") || strings.HasPrefix(objectID, "/sql/warehouses")
-}
-
-func (a PermissionsAPI) getObjectCreator(objectID string) (string, error) {
+// safePutWithOwner is a workaround for the limitation where warehouse without owners cannot have IS_OWNER set
```

Comment:
I wonder if this bug is tracked by the sql warehouse team, as this results in 2 additional functions just for this edge case